RFC/src/RFC-0170_NetworkCommunicationProtocol.md

```diff
@@ -0,0 +1,302 @@
+|---             |---                      |
+| Validator Node | Consensus Node          |
+| Base Node      | Consensus Node          |
+| Wallet         | Consensus Client        |
+| Token Wallet   | Consensus Client        |
+
+#### Unique identification of Consensus Nodes and Consensus Clients 
+
+In the Tari communication network, each CN or CC makes use of a node id to determine their position in the network.
+This node id is either assigned based on registration on the Base Layer or can be derived from the CNs or CCs identification public key.
+The method used to obtain a node id will either enhance or limit the trustworthiness of that entity when propagating messages through them on the Tari communication network.
+When performing the broadcasting of data messages or propagating discovery requests, nodes with registration assigned node ids are considered more trustworthy compared to nodes with derived node ids. 
+
+Obtaining a node id from registration on the Base Layer is important as it will limit the potential of some parties performing [Eclipse attacks](https://eprint.iacr.org/2015/263.pdf) on the network.
+Registration makes it more difficult for [Bad Actor]s to position themselves in ideal patterns on the network to perform disruptive operations and actions. 
+In sensitive situations or situations where the Kademlia-style directed propagation of messages are vulnerable, gossip protocol-based broadcasting of messages can be performed as a less efficient, but safer alternative to ensure that the message will successfully reach the rest of the network.
+
+The recommended method of node id assignment for each Tari communication network entity type are as follows:
+
+| Entity Type    | Communication Node Type | Node ID Assignment |
+|---             |---                      |---                 |
+| Validator Node | Consensus Node          | Registration       |
+| Base Node      | Consensus Node          | Derived            |
+| Wallet         | Consensus Client        | Derived            |
+| Token Wallet   | Consensus Client        | Derived            |
+
+Note that [Mining Server]s and [Mining Worker]s are excluded from the Tari communication network.
+A Mining Worker will have a local or remote P2P connection with a Mining Server.
+A Mining Server will have a local or remote P2P connection with a Base Node.
+They do not need to make use of the communication network and they are not responsible for propagating any messages on the network.
+The parent Base Node will perform any communication tasks on the Tari communication network on their behalf.
+
+#### Online Communication Address, Peer Address and Routing Table
+
+Each CC and CN on the Tari communication network will have identification cryptographic keys, a node id and an online communication address.
+The online communication address can be either an IPv4, IPv6, URL, Tor(Base32) or I2P(Base32) address and can be stored using the network address type as follows:
```

Comment:
I think this table should be added to the SHOULD requirement category.

Reply:
Change applied.